processor/src/advice/recorder.rs

```diff
@@ -0,0 +1,218 @@
+use super::{
+    AdviceInputs, AdviceProvider, AdviceSource, DataRecorder, ExecutionError, Felt, IntoBytes,
+    MerklePath, NodeIndex, RecordingMerkleStore, StarkField, Vec, Word,
+};
+
+// MEMORY ADVICE PROVIDER
+// ================================================================================================
```

Comment:
minor nit: this should probably be `RECORDING ADVICE PROVIDER`.

Reply:
updated